genai-perf/tests/test_config_command.py

```diff
@@ -0,0 +1,272 @@
+        """
+        # yapf: disable
+        yaml_str = ("""
+            model_name: gpt2
+            """)
+        # yapf: enable
+
+        user_config = yaml.safe_load(yaml_str)
+        config = ConfigCommand(user_config)
+
+        self.assertEqual(config.model_names, ["gpt2"])
+
+    def test_analyze_subcommand(self):
+        """
+        Test that the analyze subcommand is parsed correctly
+        """
+        # yapf: disable
+        yaml_str = ("""
+            model_name: gpt2
+
+            analyze:
+                concurrency:
+                    start: 4
+                    stop: 32
+
+                num_dataset_entries:
+                    start: 1
+                    stop: 5
+
+                input_sequence_length:
+                    start: 100
+                    stop: 200
+                    step: 20
+            """)
+        # yapf: enable
+
+        user_config = yaml.safe_load(yaml_str)
+        config = ConfigCommand(user_config)
+
+        self.assertEqual(config.analyze.sweep_parameters["concurrency"], Range(4, 32))
+        self.assertEqual(
+            config.analyze.sweep_parameters["num_dataset_entries"], Range(1, 5)
+        )
+        self.assertEqual(
+            config.analyze.sweep_parameters["input_sequence_length"],
+            [100, 120, 140, 160, 180, 200],
+        )
+
+    def test_endpoint(self):
+        """
+        Test that the endpoint configuration is parsed correctly
+        """
+        # yapf: disable
+        yaml_str = ("""
+            model_name: gpt2
+
+            endpoint:
+                model_selection_strategy: random
+                backend: vllm
+                custom: custom_endpoint
+                type: endpoint_type
+                service_kind: test_service
+                streaming: True
+                server_metrics_url: "test_server_metrics_url"
+                url: "test_url"
+            """)
+        # yapf: enable
+
+        user_config = yaml.safe_load(yaml_str)
+        config = ConfigCommand(user_config)
+
+        self.assertEqual(
+            config.endpoint.model_selection_strategy, ModelSelectionStrategy.RANDOM
+        )
+        self.assertEqual(config.endpoint.backend, OutputFormat.VLLM)
+        self.assertEqual(config.endpoint.custom, "custom_endpoint")
+        self.assertEqual(config.endpoint.type, "endpoint_type")
+        self.assertEqual(config.endpoint.service_kind, "test_service")
+        self.assertEqual(config.endpoint.streaming, True)
+        self.assertEqual(config.endpoint.server_metrics_url, "test_server_metrics_url")
+        self.assertEqual(config.endpoint.url, "test_url")
+
+    def test_perf_analyzer(self):
+        """
+        Test that the perf analyzer configuration is parsed correctly
+        """
+        # yapf: disable
+        yaml_str = ("""
+            model_name: gpt2
+
+            perf_analyzer:
+                path: test_path
+                stimulus:
```

Comment:
what does stimulus translate to in PA options?

Reply:
Good catch! This is either concurrency or request rate (replaces --concurrency/request-rate). I've fixed the default (should be concurrency: 1) and updated the unit testing to make this more clear.